### PR TITLE
first pass at plugin

### PR DESCRIPTION
**Pull request by pon**
@lob/api-engineering-team

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -73,12 +73,7 @@
     "===",
     "!=",
     "!=="
-
-Error: Parse error on line 75:
-..."!=",    "!=="    // etc    ],    "d
----------------------^
-Expecting 'EOF', '}', ':', ',', ']'
-// etc
+    // etc
```
Comment (leore) [inline .jscsrc]: ?

Comment (pon) [inline .jscsrc]: Just a bad copy paste, fixing it here

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  options.ignoredKeys = options.ignoredKeys || [];
+
+  try {
+    Joi.assert(options, schema, 'Invalid Configuration Option');
+  } catch (ex) {
+    return next(ex);
+  }
+
+  server.ext('onPreHandler', function (request, reply) {
+    var settings = request.route.settings.plugins.queryFilter;
```
Comment (leore) [inline lib/index.js]: @lob/api-engineering-team thoughts on queryFilter as the name. I think this is the most important feature making sure the naming convention makes the plugin intuitive

Comment (pon) [inline lib/index.js]: Made sense to me, but open to suggestions

Comment (leore) [inline lib/index.js]: I could'nt think of anything better

Comment (pon) [inline lib/index.js]: Haha same

**Comment by pon** [comment]
@lob/api-engineering-team

**Comment by leore** [comment]
LGTM my only issue is the param name. I can't think of anything better except "filter" which is more generic but might be too vague

File: .jscsrc (hunk 1 above)
Comment (mgartner) [inline .jscsrc]: Do you need this line?

**Comment by mgartner** [comment]
lgtm

**Comment by mgartner** [comment]
can we get travis out circle out of here? why does travis do two builds?

**Comment by pon** [comment]
Only travis here, not sure why it's doing multiple builds

**Pull request merged**